strongbox-data-service/src/main/java/org/carlspring/strongbox/config/janusgraph/RemoteDbServerConfiguration.java

```diff
@@ -0,0 +1,60 @@
+package org.carlspring.strongbox.config.janusgraph;
+
+import org.carlspring.strongbox.db.schema.StrongboxSchema;
+import org.janusgraph.core.JanusGraph;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.strongbox.db.server.JanusGraphConfiguration;
+import org.strongbox.db.server.JanusGraphProperties;
+import org.strongbox.db.server.JanusGraphServer;
+import org.strongbox.db.server.JanusGraphWithRemoteCassandra;
+
+/**
+ * @author sbespalov
+ */
+@Configuration
+@Conditional(RemoteDbServerConfiguration.class)
+public class RemoteDbServerConfiguration implements Condition
+{
+
+    @Bean
+    JanusGraphServer embeddedDbServer(JanusGraphConfiguration janusGraphConfiguration)
+    {
+        return new JanusGraphWithRemoteCassandra(janusGraphConfiguration);
+    }
+
+    @Bean
+    JanusGraph JanusGraph(JanusGraphServer server)
+        throws Exception
+    {
+        return new StrongboxSchema().createSchema(server.getJanusGraph());
+    }
+
+    @Override
+    public boolean matches(ConditionContext conditionContext,
+                           AnnotatedTypeMetadata metadata)
+
+    {
+        JanusGraphDbProfile profile = JanusGraphDbProfile.resolveProfile((ConfigurableEnvironment) conditionContext.getEnvironment());
+
+        return profile.getName().equals(JanusGraphDbProfile.PROFILE_REMOTE);
```

Comment:
The profile matching logic is very limiting because it assumes you will use **exactly** the `db_REMOTE.yaml` file for remote Cassandra configurations. If, like in my case, you have configured multiple clusters with different settings for testing and validation purposes you will have to go to `strongbox-db`, configure the file manually, wait a build for 1 minute, go back to the Strongbox project, build for another minute or two and only then proceed with testing. If you need to check the other cluster - you need to go back and do the same operation again which looses a lot of time. :) 

For now lets leave it as it is but if you have ideas on how to improve this - it would be an awesome follow-up task and we should discuss it in the chat.

Reply:
as an option you could override the `config-location` property with JDK options, like common Spring Environment variable. The profiles itself needed to have proper Spring context configuration, so for example with embedded profile we have cassandra instance as Spring bean.